src/packse/error.py

```diff
@@ -44,6 +44,12 @@ def __init__(self, message, output) -> None:
         super().__init__(message)
 
 
+class PublishNoCredentials(PublishError):
+    def __init__(self) -> None:
+        message = "No credentials provided for publish. Provide an API token via `PACKSE_PUBLISH_PASSWORD`."
```

Comment:
Shouldn't this be `repr` or `str`?

Reply:
I'm just matching the format for all my other error types here. I think it's often safer (for like pickle roundtrips) to pass the concrete message up to `Exception` immediately.